vignettes/messydates.Rmd

```diff
@@ -0,0 +1,151 @@
+
+```{r data, warning=FALSE}
+library(messydates)
+roman_population # For more information on the dataset please run `?roman_population`.
+```
+
+## Coerce to messydates
+
+The function `as_messydate()` handles the annotation and coercion to 'mdate' class.
+We can, for instance, coerce the date variable in the Roman population data to 'mdate'.
+
+```{r messy, warning=FALSE}
+roman_population$period <- as_messydate(roman_population$period)
+roman_population$period
+```
+
+## Annotate
+
+Some datasets have, for example, an arbitrary cut off point for start and end points, 
+but these are often coded as precise dates when they are not necessarily the real start or end dates.
+The annotate functions helps annotate uncertainty and approximation to dates.
+Inaccurate start or end dates can be represented by an ".." affix
+indicating "on or before", if used as a prefix,
+or indicating "on or after", if used as a suffix.
+Approximate dates are indicated by adding a `~` to year,
+month, or day components, as well as groups of components or whole dates
+to estimate values that are possibly correct.
+Day, month, or year, uncertainty can be indicated by adding a `?`
+to a possibly dubious date or date component.
+
+For example, we can add these annotations to the 'mdate' variable in the Roman population data.
+
+```{r annotate, warning=FALSE}
+ifelse(roman_population$period == "-0200..-0100", on_or_before(roman_population$period), roman_population$period)
+ifelse(roman_population$period == "0150..0200", on_or_after(roman_population$period), roman_population$period)
+ifelse(roman_population$period == "-0100..0001", as_approximate(roman_population$period), roman_population$period)
+ifelse(roman_population$period == "0001..0150", as_uncertain(roman_population$period), roman_population$period)
+```
+
+## Expand
+
+Expand functions transform date ranges, sets of dates, and unspecified or 
+approximate dates (annotated with '..', '{ , }', 'XX' or '~') into lists of dates.
+As these dates may refer to several possible dates,
+the function "opens" these values to include all the possible dates implied.
+
+The ranges in the 'mdate' variable in the Roman population data can be also expanded.
+
+```{r expand, warning=FALSE}
+exp <- expand(roman_population$period) # Please run `exp` to see all the expanded dates.
+lengths(exp)
+```
+
+## Contract
+
+The `contract()` function operates as the opposite of `expand()`.
+It contracts a list of dates into the abbreviated annotation of messydates.
+We can contract back the 'mdate' variable in the Roman population previously expanded.
+
+```{r contract, warning=FALSE}
+tibble::tibble(contract = contract(expand(roman_population$period), collapse = FALSE))
+```
+
+## Coerce from messydates
```

Comment:
Maybe motivate this with an example of a toy event history analysis or something?